File: mvp.admin/src/FlexForm.Admin/flexform-admin/src/app/models/dtos/form-definition.model.ts
import { FormType } from '../enums';
import { FormControl } from './form-control.model';
import { EntityDefinition } from './entity-definition.model';
import { BaseProperty } from '../properties/base-property.model';
import { UiPropertiesInterface } from '../interfaces/ui-properties.interface';
import { BaseWithTimeStamp } from './base-with-timestamp.model';
import { IViewDefinition } from '../interfaces/view-definition.interface';

export class FormDefinition extends BaseWithTimeStamp<number> implements UiPropertiesInterface, IViewDefinition{

    entityDefinitionId:number;
    formName: string;
    title: string;
    description: string;
    formType: FormType;
    isRootForm: boolean;
    isInvalidStateEnabled: boolean;
    uiProperties: BaseProperty<any>[];
    formControls:FormControl[];
    entityDefinition:EntityDefinition;
    
}

File: mvp.admin/src/FlexForm.Api.Admin/Model/FormDefinitionModel.cs
using FlexForm.DDE.Common.Enums;
using System.Collections.Generic;

namespace FlexForm.Api.Admin.Model
{
    public class FormDefinitionModel : ModelBaseInt
    {
        public int? EntityDefinitionId { get; set; }

        public string FormName { get; set; }

        public string Title { get; set; }

        public string Description { get; set; }

        public FormType FormType { get; set; }

        public bool IsRootForm { get; set; }

        public bool IsInvalidStateEnabled { get; set; }

        public dynamic UiProperties { get; set; }

        public IEnumerable<FormControlModel> FormControls { get; set; }

        public bool HasEntityInstances { get; set; }
    }
}
